src/goto-programs/validate_goto_model.h

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************\
+Module: Validate Goto Programs
+
+Author: Diffblue
+
+Date: Oct 2018
+
+\*******************************************************************/
+
+#ifndef CPROVER_GOTO_PROGRAMS_VALIDATE_GOTO_MODEL_H
+#define CPROVER_GOTO_PROGRAMS_VALIDATE_GOTO_MODEL_H
+
+#include <util/validate.h>
+
+class goto_model_validation_optionst final
+{
+public:
+  // this check is disabled by default (not all goto programs
+  // have an entry point)
+  bool entry_point_exists = false;
+
+  bool function_pointer_calls_removed = true;
+  bool check_returns_removed = true;
+  bool check_called_functions = true;
+
```

Comment:
If this is intended to be comprehensive right now, it should also check for `vector` and `complex`.  If this is intended to be a general framework then it's not super urgent.

Reply:
Its a general framework right now - everyone is invited to fill in the additional checks :-)